examples_tests/02.ComputeShader/main.cpp

```diff
@@ -1,152 +1,183 @@
 
-			auto cs_rawptr = cs.get();
-			shader = driver->getGPUObjectsFromAssets(&cs_rawptr, &cs_rawptr+1)->front();
+			compPipeline = device->createGPUComputePipeline(nullptr, core::smart_refctd_ptr(layout), std::move(cs));
 		}
-
-		compPipeline = driver->createGPUComputePipeline(nullptr, std::move(layout), std::move(shader));
 	}
 
-	auto renderObject = driver->getGPUObjectsFromAssets(&outImgViewRawPtr, &outImgViewRawPtr + 1)->front();
 
-	auto blitFBO = driver->addFrameBuffer();
-	blitFBO->attach(video::EFAP_COLOR_ATTACHMENT0, core::smart_refctd_ptr_dynamic_cast<video::IGPUImageView>(renderObject));
-
-	while (device->run())
 	{
-		driver->beginScene(true);
+		core::smart_refctd_ptr<video::IGPUCommandBuffer> cb;
+		device->createCommandBuffers(cmdpool.get(), video::IGPUCommandBuffer::EL_PRIMARY, 1u, &cb);
+		assert(cb);
+
+		cb->begin(video::IGPUCommandBuffer::EU_ONE_TIME_SUBMIT_BIT);
+
+		asset::SViewport vp;
+		vp.minDepth = 1.f;
+		vp.maxDepth = 0.f;
+		vp.x = 0u;
+		vp.y = 0u;
+		vp.width = WIN_W;
+		vp.height = WIN_H;
+		cb->setViewport(0u, 1u, &vp);
+
+		video::IGPUQueue::SSubmitInfo info;
+		auto* cb_ = cb.get();
+		info.commandBufferCount = 1u;
+		info.commandBuffers = &cb_;
+		info.pSignalSemaphores = nullptr;
+		info.signalSemaphoreCount = 0u;
+		info.pWaitSemaphores = nullptr;
+		info.waitSemaphoreCount = 0u;
+		info.pWaitDstStageMask = nullptr;
+		queue->submit(1u, &info, nullptr);
+		cb->end();
+	}
 
-		driver->bindComputePipeline(compPipeline.get());
-		const video::IGPUDescriptorSet* descriptorSet = ds0_gpu.get();
-		driver->bindDescriptorSets(video::EPBP_COMPUTE, compPipeline->getLayout(), 0u, 1u, &descriptorSet, nullptr);
-		driver->pushConstants(compPipeline->getLayout(), asset::ISpecializedShader::ESS_COMPUTE, 0u, sizeof(imgSize), imgSize);
-		driver->dispatch((imgSize[0] + 15u) / 16u, (imgSize[1] + 15u) / 16u, 1u);
+	core::smart_refctd_ptr<video::IGPUCommandBuffer> cmdbuf[SC_IMG_COUNT];
+	device->createCommandBuffers(cmdpool.get(), video::IGPUCommandBuffer::EL_PRIMARY, SC_IMG_COUNT, cmdbuf);
+	auto sc_images = sc->getImages();
+	for (uint32_t i = 0u; i < SC_IMG_COUNT; ++i)
+	{
+		auto& cb = cmdbuf[i];
+		auto& fb = fbo[i];
+
+		asset::IImage::SImageCopy region;
+		region.dstOffset = { 0, 0, 0 };
+		region.extent = { WIN_W, WIN_H, 0 };
+		cb->begin(0);
+		cb->bindDescriptorSets(nbl::asset::E_PIPELINE_BIND_POINT::EPBP_COMPUTE, layout.get(), 0, 1, (nbl::video::IGPUDescriptorSet**)&ds0_gpu.get());
+		cb->pushConstants(layout.get(), 0, 0, sizeof(uint32_t) * 2u, &core::vector2du32_SIMD(WIN_W, WIN_H));
+		cb->bindComputePipeline(compPipeline.get());
+		cb->dispatch((WIN_W + 15u) / 16u, (WIN_H + 15u) / 16u, 1u);
+		cb->copyImage(inImg.get(), nbl::asset::E_IMAGE_LAYOUT::EIL_UNDEFINED, sc_images.begin()[i].get(), nbl::asset::E_IMAGE_LAYOUT::EIL_UNDEFINED, 1, &region);
+		
+		video::IGPUCommandBuffer::SRenderpassBeginInfo info;
+		asset::SClearValue clear;
+		asset::VkRect2D area;
+		region.srcOffset = { 0, 0, 0 };
```

Comment:
??? region.srcOffset? a little too late

Reply:
I think i just gave up on that specific example when i saw it had pipeline binding issues. Will go back to it and fix

Reply:
> Will go back to it and fix

Make sure to merge my `danylo` branch first, most of it is already fixed